src/Extension/Extension.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extension;
+
+use Twig\Environment;
+use Twig\Extension\Attribute\AsTwigFilter;
+use Twig\Extension\Attribute\AsTwigFunction;
+use Twig\Extension\Attribute\AsTwigTest;
+use Twig\TwigFilter;
+use Twig\TwigFunction;
+use Twig\TwigTest;
+
+/**
+ * Abstract class for extension using the new PHP 8 attributes to define filters, functions, and tests.
+ *
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ */
+abstract class Extension extends AbstractExtension
```

Comment:
Is it better to create modify the existing `AbstractExtension`, create a new abstract class or a trait? The trait would be used with the `AbstractExtension`.

Reply:
From a DX perspective: If I used the new attributes, why would I still want to extend `AbstractExtension` directly or though this class? If we introduce those attributes, I believe we should make them work on standalone classes.

Reply:
Right. I can create a wrapper class that implement the ExtensionInterface and scan one or many object methods, passed to the constructor, to find filters/functions/tests. That would be much better. 

We need a marker to hint Symfony DI that a class contains Twig extension attributes, to avoid scanning attributes of all methods of all classes in the project.

Reply:
The marker will be `RuntimeExtensionInterface`. This interface will soon find its purpose in Twig.

Reply:
If we introduce all new attributes in this PR, can't the marker be, you know, an attribute? 🥺

Reply:
I created the `AsTwigExtension` attribute as marker.